### PR TITLE
Keyvault integration into RM

### PR DESCRIPTION
**Pull request by sushilraje**
# Description and Motivation 
We are integrating key vault in RM. Hence, in RM deployments we would not have iotHubConnectionString and storageAdapterWebServiceUrl as environment variables. Hence, a script has been added to call keyvault and set the necessary variables before in the EntryPoint of Docker container. The script is benign, as in, if Keyvault name, AppId, and AppSecret are available through environment variables, only then the script would execute.

# Change type 

- [ ] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/device-simulation-dotnet/346)

### CODE REVIEW

File: scripts/docker/content/set_env.sh [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        # export in current shell
+        export $_key=$_value
+
+        shift
+        shift
+    done
+}
+
+main() {
+  if [[ "$PCS_KEYVAULT_NAME" != "" ]] && [[ "$PCS_AAD_APPID" != "" ]] && [[ "$PCS_AAD_APPSECRET" != "" ]]; then
```
Comment (ppathan) [inline scripts/docker/content/set_env.sh]: Don't we need to change the run.cmd and run similar to webui?

File: scripts/docker/content/set_env.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
```
Comment (ppathan) [inline scripts/docker/content/set_env.sh]: My comments for this file are same as what I reviewed for webui [PR](https://github.com/Azure/pcs-remote-monitoring-webui/pull/1235)

File: scripts/docker/content/set_env.sh (hunk 1 above)
Comment (sushilraje) [reply scripts/docker/content/set_env.sh]: The run.cmd in our services/simulation is updated.

File: scripts/docker/content/set_env.sh [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+__extract_auth_server() {
+    AUTH_SERVER_URL=$(__extract_value_from_double_quotes $2)
+}
+
+#Removes "resource" prefix to extract resource type from 401 redirect of keyvault.
+__extract_resource_type() {
+    RESOURCE_TYPE=$(__extract_value_from_double_quotes $1)
+}
+
+__extract_value_from_double_quotes() {
```
Comment (ppathan) [inline scripts/docker/content/set_env.sh]: nit: Add meaningful var names.

File: scripts/docker/content/set_env.sh [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    # Remove trailing and prefixed double quotes.
+    local _temp=${string#*'"'}
+    _temp=${_temp%'"'}
+
+    #return the value
+    echo $_temp
+}
+
+__parse_json() {
+  _temp=`echo $1 | sed 's/\\\\\//\//g' | sed 's/[{}]//g' | awk -v k="text" '{n=split($0,a,","); for (i=1; i<=n; i++) print a[i]}' | sed 's/\"\:\"/\|/g' | sed 's/[\,]/ /g' | sed 's/\"//g' | grep -w $2`;
```
Comment (ppathan) [inline scripts/docker/content/set_env.sh]: nit: var names here too

Comment (sushilraje) [reply scripts/docker/content/set_env.sh]: changed

File: scripts/docker/content/set_env.sh (hunk 3 above)
Comment (sushilraje) [reply scripts/docker/content/set_env.sh]: changed

**Pull request merged**